test/gtest/ucs/test_profile.cc

```diff
@@ -164,14 +165,22 @@ void test_profile::run_profiled_code(int num_iters)
     if (num_threads() == 1) {
         profile_thread_func(&param);
     } else {
-        pthread_t threads[num_threads()];
+        std::queue<pthread_t> threads;
+
         for (int i = 0; i < num_threads(); ++i) {
-            pthread_create(&threads[i], NULL, profile_thread_func,
-                           (void*)&param);
+            pthread_t profile_thread;
+            ret = pthread_create(&profile_thread, NULL, profile_thread_func,
+                                 (void*)&param);
+            ASSERT_EQ(0, ret);
+
+            threads.push(profile_thread);
         }
-        for (int i = 0; i < num_threads(); ++i) {
+
+        while (!threads.empty()) {
```

Comment:
queue is not applicable here because it's not FIFO
the the code below, we can iterate backwards to empty the vector (back() + pop_back() ) or use std::vector<pthread_t>::iterator

Reply:
> queue is not applicable here because it's not FIFO

ah, okey. do you mean that we should `pthread_join` threads in the reverse order (i.e. `N-1 .. 0`)?
original and current codes create `0 .. N-1` and then destroy `0 .. N-1`

Reply:
yes

Reply:
done